app/main.py

```diff
@@ -1 +1,39 @@
-#  write your code here
+class Animal:
+
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
```

Comment:
no need to pass required parameters explicitly. we often do it when we're gonna pass not required parameter and there are many of them like
```
def func(param1, param2, param3, unrequired1=None, unrequired2=None, unrequired3=None, unrequired4=None, unrequired5=None, unrequired6=None):
    ...

# call function with explicit unrequired6 param so that Python knows that "Ihor" isn't a value for unrequired1
func(1, 2, 3, unrequired6="Ihor")
# instead of this
func(1, 2, 3, None, None, None, None, None, "Ihor")
```
Fix everywhere